## Bioinformatics R&D - Machine Learning Approaches on Single Cell RNA-SequencingðŸ§¬

Single-cell RNA sequencing (scRNA-seq) has become a transformative tool in understanding cellular heterogeneity. By measuring gene expression at the level of individual cells, scRNA-seq provides insights that bulk RNA-seq is incapable of, such as identification of rare or distinct cell types within a population. This capacity to capture transcriptional profiles at the single-cell level is crucial for understanding complex biological processes, including immune responses, cell differentiation, and disease mechanisms (Wolf et al., 2018). This study leverages machine learning (ML) approaches to annotate cell types within a scRNA-seq dataset from Peripheral Blood Mononuclear Cells (PBMCs).
PBMCs represents a diverse group of blood cells that include essential immune system components such as T cells, B cell, natural killer (NK) cells, and monocytes. These cells perform a range of vital immune functions, from recognising and eliminating pathogens to triggering adaptive immune responses. The dataset used in this analysis comprises of 68,000 cells, containing 4 different cell-types, providing a robust sample for developing and testing ML models (Abdelaal et al., 2019). The aim is to accurately label the cell types using computational approaches,  addressing the limitations of manual annotations. 

In this report, two machine learning models â€“ Support Vector Machine (SVM) and Neural Networks (NN) â€“ are applied to a PBMC dataset to predict cell type annotations from gene expression data. A central focus of the study is a comparison of the performance of the two models. Evaluation metrics such as accuracy, precision and recall will be used to evaluate each modelâ€™s effectiveness. SVM is suitable for high-dimensional data and handling complex classification tasks with the use of kernel functions. It is also robust to noise and is effective when there is a clear margin between classes, offering good interpretability of the class boundaries. On the contrary, NN particularly deep learning model, are better in capturing complex, non-linear relationships in the data. Given the scRNA-seq data often involves complex gene expression patterns, it is expected that the NN model will perform better than the SVM model in terms of the evaluation metrics. However, dimensionality reduction may be necessary to make NN training feasible, and this may lead to a loss of information that could negatively affect performance of the NN model. By comparing SVM and NNs, the study aims to identify the most effective approach for cell-type annotation in scRNA-seq data (Wolf et al., 2018). The findings will underscore the broader potential of machine learning in automating and refinement of scRNA-seq data analysis.

Traditionally, manual cell-type annotation in scRNA-seq relies on the expression of marker genes. For example, CD3 is a recognised marker for T cells, while CD19 is associated with B cells. Although effective for smaller datasets, this method becomes increasingly impractical for larger datasets and is susceptible to errors and biases, particularly when dealing with rare or poorly defined cell populations. To overcome these challenges, automated ML methods have become indispensable in scRNA-seq analysis (Ma et al., 2021), enabling researchers to scale up cell-type annotation while simultaneously improving the accuracy and consistency. 

ML methods automate the process of identifying cell types by detecting patterns in gene expression data. These models use labelled data â€“ cells with known identities â€“ to predict cell types in further data. ML approaches effectively enhance consistency, reduce manual labour and offer scalability for larger datasets. However, their effectiveness hinges on factors such feature selection, preprocessing steps, and the choice of algorithms (Abdelaal et al., 2019). Proper preprocessing, including normalisation and filtering, is essential to ensure the data quality, as scRNA-seq data often contains a lot of noise. 